guide/popular-topics/components-v2.md

```diff
@@ -0,0 +1,228 @@
+
+The example below shows how you can send a Thumbnail component as an Section component accessory in a channel.
+
+```js
+const { AttachmentBuilder, SectionBuilder, TextDisplayBuilder, ThumbnailBuilder, MessageFlags } = require('discord.js');
+
+const file = new AttachmentBuilder('../assets/image.png');
+
+const exampleSection = new SectionBuilder()
+	.addTextDisplayComponents(
+		new TextDisplayBuilder()
+			.setContent('This text is inside a new Text Display component! You can use **any __markdown__** available inside this component too.'),
+	)
+	.setThumbnailAccessory(
+		new ThumbnailBuilder()
+			.setDescription('ALT text displaying on the image')
+			.setURL('attachment://image.png'), // Supports arbitrary URLs such as 'https://i.imgur.com/AfFp7pu.png' as well.
+	);
+
+await channel.send({
+	components: [exampleSection],
+	files: [file],
+	flags: MessageFlags.IsComponentsV2,
+});
+```
+
+For more information how to set up custom attachments to use in your Thumbnail component URL, you can look at the guide for [attaching images in embeds](/popular-topics/embeds.md#attaching-images).
+
+### Media Gallery
+
+A Media Gallery is a content component that can display up to 10 media attachments formatted in an structured gallery. Each attachment in the Media Gallery component can have an optional ALT text (description) and can be marked as spoiler. You can use the <DocsLink path="MediaGalleryBuilder:Class" /> and <DocsLink path="MediaGalleryItemBuilder:Class" /> utility classes to easily create a Media Gallery component and its items.
+
+The example below shows how you can send a Media Gallery component in a channel.
+
+```js
+const { AttachmentBuilder, MediaGalleryBuilder, MediaGalleryItemBuilder, MessageFlags } = require('discord.js');
+
+const file = new AttachmentBuilder('../assets/image.png');
+
+const exampleGallery = new MediaGalleryBuilder()
+	.addItems(
+		new MediaGalleryItemBuilder()
+			.setDescription('ALT text displaying on an image from the AttachmentBuilder')
+			.setURL('attachment://image.png'),
+		new MediaGalleryItemBuilder()
+			.setDescription('ALT text displaying on an image from an external URL')
+			.setURL('https://i.imgur.com/AfFp7pu.png')
+			.setSpoiler(true), // Will display as a blurred image
+	);
+
+await channel.send({
+	components: [exampleGallery],
+	files: [file],
+	flags: MessageFlags.IsComponentsV2,
+});
+```
+
+### File
+
+A File is a content component that can display any uploaded file as an attachment to a message and reference it in the File component itself. It can only display 1 attachment per File component, but using multiple File components you can upload multiple files in one message. File components cannot have ALT text (description) unlike a Thumbnail or Media Gallery component, but you can add a spoiler to the component if you would like to. You can use the <DocsLink path="FileBuilder:Class" /> utility class to easily create a File component.
```

Comment:
```suggestion
A File is a content component that can display any uploaded file as an attachment to a message and reference it in the File component itself. It can only display 1 attachment per File component, but using multiple File components, you can upload multiple files in one message. File components cannot have alt text (description), unlike a Thumbnail or Media Gallery component, but you can add a spoiler to the component. You can use the <DocsLink path="FileBuilder:Class" /> utility class to easily create a File component.
```